docs/dev/stream/state/custom_serialization.md

```diff
@@ -223,15 +223,16 @@ the same `TypeSerializerSnapshot` class as their snapshot would complicate the i
 This would also be a bad separation of concerns; a single serializer's serialization schema,
 configuration, as well as how to restore it, should be consolidated in its own dedicated `TypeSerializerSnapshot` class.
 
-#### 3. Use the `CompositeSerializerSnapshot` utility for serializers that contain nested serializers
+#### 3. Extend the `CompositeTypeSerializerSnapshot` class for serializers that contain nested serializers
 
 There may be cases where a `TypeSerializer` relies on other nested `TypeSerializer`s; take for example Flink's
-`TupleSerializer`, where it is configured with nested `TypeSerializer`s for the tuple fields. In this case,
+`MapSerializer`, where it is configured with nested `TypeSerializer`s for its keys and values in the map. In this case,
 the snapshot of the most outer serializer should also contain snapshots of the nested serializers.
 
-The `CompositeSerializerSnapshot` can be used specifically for this scenario. It wraps the logic of resolving
+`CompositeTypeSerializerSnapshot` could be extended for all nested `TypeSerializer`s in this scenario. It wraps the logic of resolving
```

Comment:
This sentence is a bit strange (the "for all nested TypeSerializer" part).
I would like to stick with the original sentence ("could be used specifically for this scenario").